From 52393b256376883e8aef4f90483ab924bd03fbf3 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Mon, 4 Jan 2016 15:04:15 +0100
Subject: [PATCH 042/538] dmaengine: mv_xor: fix dummy buffers' mapping

Dummy buffers, which are used in DMA_INTERRUPT support, are dma-mapped
with NULL as a first argument. It is caused by calling dma_map_single
before dma_dev->dev assignment and may result in inappropriate DMA ops
selection.

This patch fixes the issue by replacing first argument of dma_map_single
with proper pointer.

Fixes commit 22843545b200 ("dma: mv_xor: Add support for DMA_INTERRUPT")
Cc: <stable@vger.kernel.org> v3.18+
Signed-off-by: Marcin Wojtas <mw@semihalf.com>

Change-Id: Ia12bb2d2bfc6b531949e8224f9c6fe2fffe9fb74
Reviewed-on: http://vgitil04.il.marvell.com:8080/26776
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27272
Tested-by: Lior Amsalem <alior@marvell.com>
---
 drivers/dma/mv_xor.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dma/mv_xor.c b/drivers/dma/mv_xor.c
index 14091f8..997a2b5 100644
--- a/drivers/dma/mv_xor.c
+++ b/drivers/dma/mv_xor.c
@@ -954,9 +954,9 @@ mv_xor_channel_add(struct mv_xor_device *xordev,
 	 * a DMA_INTERRUPT operation as a minimum-sized XOR operation.
 	 * Hence, we only need to map the buffers at initialization-time.
 	 */
-	mv_chan->dummy_src_addr = dma_map_single(dma_dev->dev,
+	mv_chan->dummy_src_addr = dma_map_single(&pdev->dev,
 		mv_chan->dummy_src, MV_XOR_MIN_BYTE_COUNT, DMA_FROM_DEVICE);
-	mv_chan->dummy_dst_addr = dma_map_single(dma_dev->dev,
+	mv_chan->dummy_dst_addr = dma_map_single(&pdev->dev,
 		mv_chan->dummy_dst, MV_XOR_MIN_BYTE_COUNT, DMA_TO_DEVICE);
 
 	/* allocate coherent memory for hardware descriptors
-- 
1.9.1

